internal/storage-service/service.go

```diff
@@ -206,6 +206,77 @@ func (s *Service) Update(ctx context.Context, req *pb.StorageUpdateRequest) (*pb
 	return &pb.StorageUpdateResponse{}, nil
 }
 
+// Delete deletes a storage
+func (s *Service) Delete(ctx context.Context, req *pb.StorageDeleteRequest) (*pb.StorageDeleteResponse, error) {
+	s.log.WithFields(logrus.Fields{
+		"StorageType": req.StorageType,
+		"SystemId":    req.SystemId,
+	}).Info("Serving delete storage request")
+
+	// Get the current list of registered storage systems
+	s.log.Debug("Getting existing storages")
+	existingStorages, err := s.kube.GetConfiguredStorage(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	s.log.Debug("Getting system type")
+	systemType, ok := existingStorages[req.StorageType]
+	if !ok {
+		return nil, fmt.Errorf("error: storage of type %s is missing", req.StorageType)
+	}
+
+	s.log.Debug("Check the requested system ID exists")
+	if _, systemIDExists := systemType[req.SystemId]; !systemIDExists {
+		return nil, fmt.Errorf("error: system with ID %s does not exist", req.SystemId)
+	}
+
+	delete(systemType, req.SystemId)
+	existingStorages[req.StorageType] = systemType
+	err = s.kube.UpdateStorages(ctx, existingStorages)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.StorageDeleteResponse{}, nil
+}
+
+// Get retrieves a storage info
+func (s *Service) Get(ctx context.Context, req *pb.StorageGetRequest) (*pb.StorageGetResponse, error) {
+	s.log.WithFields(logrus.Fields{
+		"StorageType": req.StorageType,
+		"SystemId":    req.SystemId,
+	}).Info("Serving get storage request")
+
+	// Get the current list of registered storage systems
+	s.log.Debug("Getting existing storages")
+	existingStorages, err := s.kube.GetConfiguredStorage(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	s.log.Debug("Getting system type")
+	systemType, ok := existingStorages[req.StorageType]
+	if !ok {
+		return nil, fmt.Errorf("error: storage of type %s is missing", req.StorageType)
+	}
+
+	s.log.Debug("Check the requested system ID exists")
+	if _, systemIDExists := systemType[req.SystemId]; !systemIDExists {
+		return nil, fmt.Errorf("error: system with ID %s does not exist", req.SystemId)
+	}
+
+	s.log.Debug("JSON marshaling configured storage")
+	system := systemType[req.SystemId]
+	system.Password = "(omitted)"
```

Comment:
The password is scrubbed on the client (karavictl) side. I'd suggest to remove this line or the one in karavictl.

https://github.com/dell/karavi-authorization/blob/1c6ec905cfcbc5c035df599d91716fecf24c8ee8/cmd/karavictl/cmd/storage_get.go#L104

Reply:
Per discussion, the karavictl code is necessary for the rpm functionality.